.changeset/config.json

```diff
@@ -0,0 +1,7 @@
+{
+  "$schema": "https://unpkg.com/@changesets/config@0.2.1/schema.json",
+  "changelog": "@changesets/cli/changelog",
+  "commit": true,
```

Comment:
Looking through docs I still am not sure what exact result does it have - for both cases. Could you describe in a little bit more detail how this works?

Reply:
Making a note to clarify in the docs, but short summary here:

If `commit` is true:

- when you run the `changeset` command, it will, after adding the changeset file, stage and commit for you.
- when you run the version command, after updating the changelogs and versions, it will stage and commit for you.

If `commit` is false, changesets will never stage or commit files.

Reply:
It would be good to have the ability to configure this separately - it seems like I wouldn't like creating changesets to create commits, but I would see myself wanting auto-commits after versioning. 

Also - just a sanity-check - do you verify if there are any uncommitted changes before running version & publish? Seems like it would be good to fail in such situation.

Reply:
> It would be good to have the ability to configure this separately

We had that in v1, and almost removed the `commit` option entirely, but ended up deciding to move to one option.

If you are opting into auto-versioning through the GH action, you don't run the versioning command yourself. Will add this to our documentation around recommendations.

> do you verify if there are any uncommitted changes before running version & publish?

We verify there are no uncommitted changes before making any commit, but not explicitly before version and publish. Changesets theoretically works outside of git, and we've been trying to reduce the print of where we touch git.

I'd probably recommend making that part of any publish script you run (you generally want something like `yarn publish` instead of running `yarn changeset publish` just so you can make a script of `yarn build && changeset publish`)

Reply:
Err, only, not `yarn publish` but `yarn release` because naming a script publish is a massive footgun I have done before.

Reply:
> We verify there are no uncommitted changes before making any commit, but not explicitly before version and publish. Changesets theoretically works outside of git, and we've been trying to reduce the print of where we touch git.

Well - it seems to me changesets are pretty much coupled to git querying anyway, so don't get that argument here, but I also don't care that much as I intend to automate things with changesets and in such scenario I can't land with uncommitted stuff before those actions (unless my scripts are broken, or I don't .gitignore stuff properly).

As to `changeset publish` - does it run `changeset version` implicitly as well? From what I see the bot doesn't have to be configured to run version command separately.

Reply:
> If you are opting into auto-versioning through the GH action, you don't run the versioning command yourself. 

That's right - but even auto-versioning needs to decide if it should do extra commits or not, right? I would expect this to affect the automated process as well.

Reply:
If you're doing auto-versioning we have to commit the changes no matter what your config, otherwise it couldn't raise the PR.

Reply:
To raise an issue - yeah, but I'm still a little bit unsure what happens in regards to git when the actual release is being done. We merge a PR so obviously the change lands in git, this triggers the action - it runs versioning command (which probably creates git tags etc) and also runs publish command, but I'm not sure if it this creates a separate commit from the one that has landed into master? And not sure how commit setting applies to that process when taking auto-versioning into account.

Reply:
An important clarification on what the action does is that it will only run _one of_ version or publish: If there are changesets on master, it will run version, commit to a branch and open a PR. If there aren't changesets on master, it will run publish which will publish to npm and create the git tags but it doesn't modify anything on the file system/there isn't anything that needs to be committed.

Reply:
Ok, this explains it a bit more for me 👍 I was missing the part that versioning actually already happens on the created branch.

One additional thing I got from this and reading the action's source code - the `commit` option is relevant only when using `changeset` tool by hand. It doesn't affect bot and the action anyhow. Might be good to clarify this in the docs @Noviny